samza-core/src/main/java/org/apache/samza/util/FutureUtil.java

```diff
@@ -144,31 +141,21 @@ public static <K, V> CompletableFuture<Map<K, V>> toFutureOfMap(
 
   public static <T> CompletableFuture<T> executeAsyncWithRetries(String opName,
       Supplier<? extends CompletionStage<T>> action, Predicate<? extends Throwable> abortRetries,
-      ExecutorService executor) {
-    Duration maxDuration = Duration.ofMinutes(10);
+      ExecutorService executor, RetryPolicyConfig retryPolicyConfig) {
 
     RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
-        .withMaxRetries(-1) // Sets maximum retry to unlimited from default of 3 attempts. Retries are now limited by max duration and not retry counts.
-        .withBackoff(100, 312500, ChronoUnit.MILLIS, 5) // 100 ms, 500 ms, 2500 ms, 12.5 s, 1.05 min, 5.20 min, 5.20 min
-        .withMaxDuration(maxDuration)
-        .withJitter(Duration.ofMillis(100))
-        .abortOn(abortRetries) // stop retrying if predicate returns true
-        .onRetry(e -> LOG.warn("Action: {} attempt: {} completed with error {} ms after start. Retrying up to {} ms.",
-            opName, e.getAttemptCount(), e.getElapsedTime().toMillis(), maxDuration.toMillis(), e.getLastFailure()));
-
-    return executeAsyncWithRetries(opName, action, abortRetries, executor, retryPolicy);
-  }
-
-  public static <T> CompletableFuture<T> executeAsyncWithRetries(String opName,
-      Supplier<? extends CompletionStage<T>> action, Predicate<? extends Throwable> abortRetries,
-      ExecutorService executor, RetryPolicy<Object> retryPolicy) {
-
-    RetryPolicy<Object> retryPolicyWithLog = retryPolicy.abortOn(abortRetries)
+        .withMaxRetries(retryPolicyConfig.getMaxRetries())
```

Comment:
Nvm, saw below that this is a custom class. Looks like there is an existing RetryPolicyConfig class in Failsafe [1]. Might only be in a newer version. Check if it's possible to upgrade. Otherwise LGTM for time being.

[1] https://failsafe.dev/javadoc/core/dev/failsafe/RetryPolicy.html#builder-dev.failsafe.RetryPolicyConfig-

Reply:
Yes, RetryPolicyConfig is not available yet in the [latest release version](https://repo.maven.apache.org/maven2/net/jodah/failsafe/2.4.4/) of FailSafe library.